Model/Configuration.php

```diff
@@ -116,4 +152,34 @@ protected function getStoreConfig($path)
             $this->storeId
         );
     }
+
+		/**
+		 * @param string $path
+		 * @param mixed $val
+		 */
+		protected function setStoreConfig($path, $val)
+		{
+				$this->resourceConfig->saveConfig(
+						$path,
+						$val,
+						$this->scope,
+						$this->storeId
+				);
+				$this->storeManager->getStore($this->storeId)->resetConfig();
+		}
+
+		/**
+		 * @param string $path
+		 * @param mixed $val
+		 */
+		protected function setWebsiteConfig($path, $val)
+		{
+				$this->resourceConfig->saveConfig(
+						$path,
+						$val,
+						\Magento\Store\Model\ScopeInterface::SCOPE_WEBSITES,
+						$this->storeId
```

Comment:
This won't work… Store ID and website ID aren't the same, so this'll return invalid data.

Might be worth fixing in a different PR, however, where we switch over to just having everything be based on the website scope instead of store view.